### PR TITLE
[Docs] Allow building man pages without myst_parser

### PR DESCRIPTION
**Pull request by cor3ntin**
The man pages do not depend on the doc present in markdown files, so they can be built without myst_parser.
Doing so might allow llvm distributions to have less development dependencies.

As we do not have the ennvironment to test these configuration, this capability is provided on a best-effort basis.

This restores a capability accidentally lost in #65664.

### CODE REVIEW

**Review by AaronBallman — COMMENTED** [review]
These changes look reasonable to me, but I'd appreciate a final sign off from @mgorny to make sure this works for his needs.

File: clang/docs/conf.py [hunk 1]
```diff
@@ -35,8 +35,16 @@
 
 import sphinx
 
-if sphinx.version_info >= (3, 0):
-    extensions.append("myst_parser")
+# When building man pages, we do not use the markdown pages,
+# So, we can continue without the myst_parser dependencies.
+# Doing so reduces dependencies of some packaged llvm distributions.
+try:
+  import myst_parser
```
Comment (mgorny) [inline clang/docs/conf.py:42]: I think you're using different indentation (2 spaces) vs other code in that file (4 spaces).

**Review by mgorny — APPROVED** [review]
Thank you. From a quick test run with and without `myst-parser`, this seems to work as intended.

**Review by AaronBallman — APPROVED** [review]
LGTM aside from the formatting issue @mgorny mentioned.

**Pull request merged**